File: backend/src/main/java/org/akashihi/mdg/api/v1/json/TagDeserializer.java
package org.akashihi.mdg.api.v1.json;

import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
import org.akashihi.mdg.entity.Tag;

import java.io.IOException;

public class TagDeserializer extends StdDeserializer<Tag> {
    protected TagDeserializer() {
        this(null);
    }

    protected TagDeserializer(Class<?> vc) {
        super(vc);
    }

    @Override
    public Tag deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException {
        var tagString = deserializationContext.readValue(jsonParser, String.class);
        return new Tag(tagString);
    }
}

File: backend/src/main/kotlin/org/akashihi/mdg/api/v1/json/TagDeserializer.kt
package org.akashihi.mdg.api.v1.json

import com.fasterxml.jackson.core.JsonParser
import com.fasterxml.jackson.databind.DeserializationContext
import com.fasterxml.jackson.databind.deser.std.StdDeserializer
import org.akashihi.mdg.entity.Tag
import java.io.IOException

open class TagDeserializer protected constructor(vc: Class<*>? = null) : StdDeserializer<Tag>(vc) {
    @Throws(IOException::class)
    override fun deserialize(jsonParser: JsonParser, deserializationContext: DeserializationContext): Tag {
        val tagString = deserializationContext.readValue(jsonParser, String::class.java)
        return Tag(tagString)
    }
}
